src/inet/InetInterface.cpp

```diff
@@ -502,6 +502,142 @@ void CloseIOCTLSocket()
     }
 }
 
+#if __ANDROID__
+
+static struct if_nameindex * backport_if_nameindex(void);
```

Comment:
Is this because Android does not implement `if_nameindex`?

I do have to ask: should we just have an Android-only file that implements `if_nameindex` and `if_freenameindex` so this core code doesn't have to have the ifdefs?

Reply:
Is this because Android does not implement if_nameindex?
 -> Yes
I do have to ask: should we just have an Android-only file that implements if_nameindex and if_freenameindex so this core code doesn't have to have the ifdefs?
 -> The reason I just did this was because this code had previously been deleted, so I restored it. 
     #25856 
     It is not easy to change this code to be implemented in the Android folder.

Reply:
>  It is not easy to change this code to be implemented in the Android folder.

I wasn't asking for that.  I was asking for a separate file that is compiled only on Android.  It can live in this directory.

I didn't realize we had used to have this workaround in this form, but the question remains: if the issue is that `if_nameindex` and `if_freenameindex` don't exist, then why are we not naming our implementations _that_, and then the cross-platform code can just work without needing the Android ifdefs.

Reply:
@joonhaengHeo see above.